python/pyiceberg/expressions/literals.py

```diff
@@ -86,6 +86,8 @@ def __hash__(self) -> int:
         return hash(self.value)
 
     def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, Literal):
```

Comment:
If we get something else, we could wrap it in a literal and compare it. This would allow `literal(1) == 1`.

Reply:
I don't think that we want to do that. Then equality would not be symmetric.